kbs/src/jwe.rs

```diff
@@ -43,23 +53,65 @@ pub fn jwe(tee_pub_key: TeePubKey, payload_data: Vec<u8>) -> Result<Response> {
 
     let rsa_pub_key =
         RsaPublicKey::new(n, e).context("Building RSA key from modulus and exponent failed")?;
-    let sym_key: &[u8] = aes_sym_key.as_slice();
-    let wrapped_sym_key = rsa_pub_key
-        .encrypt(&mut rng, Pkcs1v15Encrypt, sym_key)
+    let encrypted_key = rsa_pub_key
+        .encrypt(&mut rng, Pkcs1v15Encrypt, aes_sym_key.as_slice())
         .context("RSA encrypt sym key failed")?;
 
-    let protected_header = json!(
-    {
-       "alg": RSA_ALGORITHM.to_string(),
-       "enc": AES_GCM_256_ALGORITHM.to_string(),
-    });
-
     Ok(Response {
-        protected: serde_json::to_string(&protected_header)
-            .context("serde protected_header failed")?,
-        encrypted_key: URL_SAFE_NO_PAD.encode(wrapped_sym_key),
-        iv: URL_SAFE_NO_PAD.encode(iv),
-        ciphertext: URL_SAFE_NO_PAD.encode(encrypted_payload_data),
-        tag: "".to_string(),
+        protected,
+        encrypted_key,
+        iv: iv.into(),
+        ciphertext: payload_data,
+        aad: None,
+        tag: tag.to_vec(),
     })
 }
+
+#[cfg(test)]
+mod tests {
+    use core::assert_eq;
+
+    use base64::{engine::general_purpose::URL_SAFE_NO_PAD, Engine};
+    use josekit::jwe::{alg::rsaes::RsaesJweAlgorithm::Rsa1_5, JweContext, JweHeader};
+    use kbs_types::TeePubKey;
+    use openssl::rsa::Rsa;
+
+    use super::jwe;
+
+    #[test]
+    fn jwe_compability() {
```

Comment:
@mkulke Following your nice idea, I added this unit test and found some potential bugs of serialization of jwe. Now fixed in upstream and this unit test passes locally on my side. 

cc @tylerfanelli with https://github.com/virtee/kbs-types/pull/53 we could make the response fully compatible with `josekit` crate which is rust crate that facilitates `jwe`.